core/arch/arm/mm/core_mmu.c

```diff
@@ -1944,3 +1944,21 @@ vaddr_t core_mmu_get_va(paddr_t pa, enum teecore_memtypes type)
 
 	return (vaddr_t)pa;
 }
+
+bool is_unpaged_code(void *va)
+{
+	paddr_t pa = virt_to_phys(va);
+
+	return pa && phys_to_virt(pa, MEM_AREA_TEE_RAM_RX) == va;
```

Comment:
This isn't accurate since `phys_to_virt()` will use `phys_to_virt_tee_ram()` for the lookup.

I don't see how it's necessary for debug purposes to check if something resides in data or code section, it seems you're trying to catch a problem that only can occur if created deliberately. However, I agree that it quite friendly to check that the function/data resides in the unpaged section as it's quite easy to get that wrong by mistake.

So checking that the virtual address is in the the range TEE_LOAD_ADDR..get_linear_map_end() in the page case should be enough.